modules/data.land/R/soilgrids_soc_extraction.R

```diff
@@ -132,22 +167,29 @@ soilgrids_soilC_extract <- function (site_info, outdir=NULL, verbose=TRUE) {
   }
   
   fitQ <- function(x) {
-    val = x$Value
-    stat = as.character(x$Quantile)
-    theta = c(10, 10)
-    fit <-
-      list(Gamma = stats::optim(theta, cgamma, val = val, stat = stat))
-    SS <- sapply(fit, function(f) {
-      f$value
-    })
-    par <- sapply(fit, function(f) {
-      f$par
-    })
-    return(list(par = par, SS = SS))
+    val <- x$Value
+    stat <- as.character(x$Quantile)
+    # Skip fitting if all values are NA or not numeric
+    if (all(is.na(val)) || length(val) == 0) {
+      return(list(par = c(NA, NA), SS = NA))
+    }
+    theta <- c(10, 10)
+    fit <- tryCatch(
+      stats::optim(theta, cgamma, val = val, stat = stat),
+      error = function(e) NULL
+    )
+    if (is.null(fit)) {
+      return(list(par = c(NA, NA), SS = NA))
+    }
+    return(list(par = fit$par, SS = fit$value))
```

Comment:
What are the possible lengths of x here? If it's always 1 I'm confused why line 173 is needed, if it can be more than 1 then I don't think list(par = par$fit,...) and list(par = sapply(fit, \(f)x$par), ...) are equivalent